KifuSF/CreateDonationViewController.swift

```diff
@@ -0,0 +1,246 @@
+        contentScrollView.updateBottomPadding(pickUpAddressButton.frame.height + 24)
+    }
+
+    @objc private func dismissVC() {
+        dismiss(animated: true)
+    }
+
+    @objc private func dismissKeyboard() {
+        view.endEditing(true)
+    }
+
+    @objc private func imageViewTapped() {
+        imageHelper.presentActionSheet(from: self)
+    }
+
+    @objc private func pickUpAddressButtonTapped() {
+        guard let _ = userSelectedAProfileImage,
+            let image = descriptorView.imageView.image,
+            let title = titleInputView.contentView.textField.text, !title.isEmpty,
+            let description = descriptionInputView.contentView.text, description != CreateDonationViewController.descriptionPlaceholder else {
+                return showErrorMessage("Please complete all the fields")
+        }
+
+        let locationPicker = LocationPickerViewController()
+
+        locationPicker.showCurrentLocationInitially = true
+        locationPicker.searchBarPlaceholder = "Choose Pickup location"
+        locationPicker.mapType = .standard
+
+        locationPicker.showCurrentLocationButton = true
+
+        locationPicker.completion = { location in
+            guard let location = location else {
+                return print("no location selected")
+            }
+
+            self.pickupLocation = location
+        }
+
+        self.navigationController?.pushViewController(locationPicker, animated: true)
+    }
+
+    private func showErrorMessage(_ errorMessage: String) {
+        errorLabel.isHidden = false
+        errorLabel.text = errorMessage
+
+        UIView.animate(withDuration: UIView.microInteractionDuration, animations: { [unowned self] in
+            self.view.layoutIfNeeded()
+        }, completion: { [unowned self] (_) in
+            self.contentScrollView.scrollToBottom()
+        })
+
+        pickUpAddressButton.resetState()
+    }
+}
+
+//MARK: - UITextFieldDelegate
+extension CreateDonationViewController: UITextFieldDelegate {
+    func textFieldShouldReturn(_ textField: UITextField) -> Bool {
+        descriptionInputView.contentView.becomeFirstResponder()
+
+        return true
+    }
+}
+
+//MARK: - UITextViewDelegate
+extension CreateDonationViewController: UITextViewDelegate {
+    func textViewDidBeginEditing(_ textView: UITextView) {
+        if textView.textColor == .kfPlaceholderText {
+            textView.textColor = .black
+            textView.text = nil
+        }
+    }
+
+    func textViewDidEndEditing(_ textView: UITextView) {
+        if textView.text.isEmpty {
+            textView.textColor = .kfPlaceholderText
+            textView.text = CreateDonationViewController.descriptionPlaceholder
+        }
+    }
+
+    func textView(_ textView: UITextView, shouldChangeTextIn range: NSRange, replacementText text: String) -> Bool {
+        if (text == "\n") {
+            textView.resignFirstResponder()
+            pickUpAddressButtonTapped()
+        }
```

Comment:
I believe this prevents the user from adding multiple lines in the notes section. Do we want that behavior? If not, we could add a `.inputAccessoryView` to the `UITextView` so we can display a done button.

Reply:
I have already set the returnKeyType to done and as you said this prevents the user from adding a new line to description.

Reply:
If this is what we want for the user, then sounds good!